app/src/main/java/com/example/pokemonultimate/ui/screens/inscription/InscriptionScreen.kt

```diff
@@ -0,0 +1,372 @@
+    val password by viewModel.password
+    val confirmPassword by viewModel.confirmPassword
+    val errorMessage by viewModel.errorInscriptionMessage
+
+    val showBottomSheet by viewModel.showBottomSheet
+
+    LocalContext.current
+
+
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .verticalScroll(rememberScrollState()),
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+
+
+        PikachuWithButtonPlus(viewModel, { viewModel.toggleBottomSheet() })
+
+        WelcomeText()
+
+        // Utilisation de CustomTextField pour le mail utilisateur
+        CustomTextField(
+            label = stringResource(id = R.string.email),
+            value = email,
+            onValueChange = { viewModel.onEmailChanged(it) },
+            leadingIconRes = R.drawable.ic_user,
+            isPasswordField = false
+        )
+
+        // Utilisation de CustomTextField pour le mot de passe
+        CustomTextField(
+            label = stringResource(id = R.string.password),
+            value = password,
+            onValueChange = { viewModel.onPasswordChanged(it) },
+            leadingIconRes = R.drawable.ic_password,
+            isPasswordField = true
+        )
+
+        // Utilisation de CustomTextField pour la confirmation du mot de passe
+        CustomTextField(
+            label = stringResource(id = R.string.confirm_password),
+            value = confirmPassword,
+            onValueChange = { viewModel.onConfirmPasswordChanged(it) },
+            leadingIconRes = R.drawable.ic_password,
+            isPasswordField = true
+        )
+
+        Box(
+            modifier = Modifier
+                .fillMaxWidth()
+                .height(48.dp),
+            contentAlignment = Alignment.Center
+        ) {
+            Text(
+                text = errorMessage,
+                color = MaterialTheme.colorScheme.error,
+                textAlign = TextAlign.Center,
+                modifier = Modifier.padding(horizontal = Padding.HUGE.dp, vertical = 0.dp)
+            )
+        }
+
+
+        ButtonSignup(viewModel = viewModel) { isSuccess ->
+            if (isSuccess) {
+                onSuccess()
+            }
+        }
+
+
+        if (showBottomSheet) BottomBar(viewModel)
+        OrView()
+        GoogleView()
+        SigninView(navController)
+    }
+}
+
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun BottomBar(viewModel: InscriptionViewModel) {
+
+    val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = false)
+
+    ModalBottomSheet(
+        modifier = Modifier,
+        sheetState = sheetState,
+        onDismissRequest = { viewModel.closeBottomSheet() },
+    ) {
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(horizontal = Padding.MEDIUM.dp),
+            contentAlignment = Alignment.Center
+        ) {
+            LazyVerticalGrid(
+                columns = GridCells.Fixed(3),
```

Comment:
maybe it should not be Fixel but Adaptative so the Bottom Sheet could adapt to the size of the screen